### ISSUE TITLE
OSX:  use *.dylib for lib extension not *.so

### ISSUE THREAD

**Issue by gaming-hacker** [flame/blis#153]
All,

OSX uses *.dylib for shared libraries not *.so.  Please fix your makefile.
currently

BLIS_DLL_NAME      := $(BLIS_LIB_BASE_NAME).so

easy fix
OSX := $(shell uname)
ifeq ($(OSX),Darwin)
BLIS_DLL_NAME      := $(BLIS_LIB_BASE_NAME).dylib
else
BLIS_DLL_NAME      := $(BLIS_LIB_BASE_NAME).so
endif

Thanks.

**Comment by devinamatthews** [comment]
.so files (at least the ways BLIS builds them) work fine on OSX on all versions that I have tested. True this would probably be an easy fix but nothing is actually broken at the moment. As an example I linked the testsuite with libblis.so no problem:

```
[dmatthews@Devin's iMac release]$rm lib/haswell/libblis.a
[dmatthews@Devin's iMac release]$gcc -o test_libblis.x obj/haswell/testsuite/*.o -Llib/haswell -lblis
[dmatthews@Devin's iMac release]$otool -L ./test_libblis.x 
./test_libblis.x:
	lib/haswell/libblis.so (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
	/usr/local/lib/gcc/7/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
[dmatthews@Devin's iMac release]$otool -hv lib/haswell/libblis.so 
Mach header
      magic cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
MH_MAGIC_64  X86_64        ALL  0x00       DYLIB    14       1984   NOUNDEFS DYLDLINK TWOLEVEL NO_REEXPORTED_DYLIBS
```

**Comment by fgvanzee** [comment]
@gaming-hacker This issue has been addressed in af1d847. Thanks for your patience, and sorry it took us so long to get around to this.

Please give it a try and let us know if the changes are satisfactory.
